apps/test/tickWrapperTest.js

```diff
@@ -0,0 +1,79 @@
+var tickWrapper = require('./util/tickWrapper');
+var assert = require('./util/testUtils').assert;
+
+function createFakeApp() {
+  return {
+    tickCount: 0,
+    onTick: function () {
+      console.log("actual");
+      this.tickCount++;
+    }
+  };
+}
+
+describe('tickWrapper', function () {
+  afterEach(function () {
+    tickWrapper.reset();
+  });
+
+  it('runOnAppTick', function () {
+    var app1 = createFakeApp();
+
+    var calledMe = false;
+
+    tickWrapper.runOnAppTick(app1, 2, function () {
+      calledMe = true;
+    });
+
+    app1.onTick();
+    assert.equal(calledMe, false);
+    assert.equal(app1.tickCount, 1);
+
+    app1.onTick();
+    assert.equal(calledMe, false);
+    assert.equal(app1.tickCount, 2);
+
+    app1.onTick();
+    assert.equal(calledMe, true);
+    assert.equal(app1.tickCount, 3);
+  });
+
+  it('tickAppUntil', function (done) {
+    var app1 = createFakeApp();
+
+    tickWrapper.tickAppUntil(app1, function () {
+      console.log("predicate");
+      return app1.tickCount === 3;
+    }).then(function () {
+      // tickCount is 4 because our predicate function runs at the beginning of
+      // the loop, and our original onTick is still called before we get to
+      // promise resolution
+      assert.equal(app1.tickCount, 4);
+      done();
+    });
+
+    app1.onTick();
+    app1.onTick();
+    app1.onTick();
+    app1.onTick();
+  });
+
+  it('reset', function () {
+    var app1 = createFakeApp();
+    var originalOnTick = app1.onTick;
+
+    var calledMe = false;
+    tickWrapper.runOnAppTick(app1, 2, function () {
```

Comment:
I see in your tickWrapper implementation that you've done the work to support multiple wrapper functions correctly, but your tests here never exercise that case.  Will you add a test that shows you can wrap twice, reset once, and get back in a good state?